foundry/foundry.py

```diff
@@ -208,12 +208,15 @@ def load(self, name, download=True, globus=True, verbose=False, metadata=None, a
 
         return self
 
-    def list(self):
+    def list(self, name_filter: Optional[Callable[[str], bool]] = None, year_filter: Optional[Callable[[np.int64], bool]] = None):
         """List available Foundry data packages
+        Args:
+            name_filter (callable): A function or lambda which takes in the name of the dataset and returns a boolean; if None, not used
+            year_filter (callable): A function or lambda which takes in the year of publication of the dataset and returns a boolean; if None, not used
 
         Returns
         -------
-            (pandas.DataFrame): DataFrame with summary list of Foundry data packages including name, title, publication year, and DOI
+            (pandas.DataFrame): DataFrame with summary list of Foundry data packages including name, title, publication year, and DOI, given the filters.
         """
         res = (
             self.forge_client.match_field(
```

Comment:
Hi @Aadit-Ambadkar ! I think instead of filtering after the fetch, we want to use Forge and Globus SDK methods to search the database appropriately. Otherwise, when we eventually have 1000s of datasets, it'll be inefficient to postprocess (and also, it would be nice for the search parameters to be a little more flexible). @blaiszik has already started working on this issue, so I think it would be best to let him finish it.

Reply:
Pandas is reliably (like ridiculously) fast, potentially faster than using Forge / Globus SDK methods, but I think it makes sense.